hackathon_site/dashboard/frontend/src/components/general/SideSheetR/SideSheetR.test.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import SideSheetR from "./SideSheetR";
+import { render, fireEvent, waitForElementToBeRemoved } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+/**
+ * Test the quantity log thing
+ */
+
+describe("SideSheetR", () => {
+    const testDetail = {
+        name: "Arduino",
+        type: "red",
+        total: 30,
+        available: 29,
+        img: "https://www.filterforge.com/more/help/images/size200.jpg",
+        tags: "MCU, FPGA",
+        manufacturer: "Canakit",
+        model_num: "Model 3B+",
+        datasheet: "link",
+        notes: ["- For micropython ask for image", "- randomnerdtutorials.com"],
+        constraints: ["- Max 1 of this item", "- Max 3 microcontroller labelled red"],
+        quantity: 3,
+    };
+
+    const addCartMock = jest.fn();
+
+    test("render", () => {
+        const { asFragment } = render(
+            <SideSheetR detail={testDetail} cart={addCartMock} />
+        );
+
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    test("click me opens the drawer", () => {
+        const { getByText, queryByText } = render(
+            <SideSheetR detail={testDetail} cart={addCartMock} />
+        );
+
+        //Open Drawer
+        const openDrawer = getByText("click me");
+        fireEvent.click(openDrawer);
+
+        // Confirm open drawer
+        expect(queryByText("Product Overview")).not.toBeNull();
+    });
+
+    test("close the drawer works", async () => {
+        const { getByText, getByRole, queryByText } = render(
+            <SideSheetR detail={testDetail} cart={addCartMock} />
+        );
+        //Open Drawer
+        const openDrawer = getByText("click me");
+        fireEvent.click(openDrawer);
+
+        const returnArrow = getByRole("close");
+        fireEvent.click(returnArrow);
+
+        await waitForElementToBeRemoved(() => getByText("Product Overview"));
```

Comment:
Great use of `waitForElementToBeRemoved`!

Could you flesh this test out to have an `expect(...).toBeInTheDocument()`? Technically this will work though, since `getByText` will throw an error if it's not there - so it's up to you if you want to change this for consistency.

Reply:
So replace `getByText` with `expect()toBeInTheDocument()`? That gives me an error saying the element you are waiting to be removed has already been removed.